gateway-ha/src/main/java/io/trino/gateway/ha/security/LbTokenUtil.java

```diff
@@ -28,11 +30,16 @@ public static boolean validateToken(String idToken, RSAPublicKey publicKey, Stri
       }
 
       Algorithm algorithm = Algorithm.RSA256(publicKey, null);
-      JWT.require(algorithm)
+      Verification verification =
+          JWT.require(algorithm)
           .withIssuer(issuer)
-          .acceptLeeway(60 * 60) // Expired tokens are valid for an hour
-          .build()
-          .verify(idToken);
+          .acceptLeeway(60 * 60); // Expired tokens are valid for an hour
+
+      if(audiences != null){
```

Comment:
Trino codestyle prefers using `Optional` to handling nulls explicitly, so the signature should be changed to `Optional<List<String>> audiences` and the check here should be `audiences.isPresent()`.